vim-patch:8.0.1774,8.1.{192,194,255,264,1202,2172,2340},8.2.{62,71,72,1004,1254,1259,1265,1267,1292,1295}

janlazo opened the pull request:
Patches from https://github.com/neovim/neovim/pull/12643

Why is there a flaky data race for the `64 bytes` terminal test on FreeBSD?

Need help reviewing patch 8.1.0192. I overdid the lint changes.

janlazo committed:
vim-patch:8.1.2172: spell highlight is wrong at start of the line

Problem:    Spell highlight is wrong at start of the line.
Solution:   Fix setting the "v" variable. (closes vim/vim#5078)
https://github.com/vim/vim/commit/7751d1d1a3d447d0d48f57f34e0da9f7c6ac433d

Skip spell tests in OpenBSD.

Nvim or screen likely crashed.
Revist once issue #12104 is fixed.
Skip the test for the following reasons:
- unknown regression caused by https://github.com/neovim/neovim/issues/12104
- cannot revert failing test from https://github.com/neovim/neovim/commit/ed0d135247790f67ead7f1a8c21a020790e80a74

janlazo committed:
vim-patch:8.0.1774: reading very long lines can be slow

Problem:    Reading very long lines can be slow.
Solution:   Read up to 1 Mbyte at a time to avoid a lot of copying.  Add a
            check for going over the column limit.
https://github.com/vim/vim/commit/13d3b05ed2cf9a54b18b4e8236f0af2c5386200c

janlazo committed:
vim-patch:8.2.1004: line numbers below filler lines not always updated

Problem:    Line numbers below filler lines not always updated.
Solution:   Don't break out of the win_line() loop too early. (Christian
            Brabandt, closes vim/vim#6294, closes vim/vim#6138)
https://github.com/vim/vim/commit/511feec6f0a9e954f1d7353425fa51232b1a8e88

janlazo committed:
vim-patch:8.2.0072: memory test still fails on Cirrus CI

Problem:    Memory test still fails on Cirrus CI.
Solution:   Allow for a tiny bit more tolerance in the upper limit.
https://github.com/vim/vim/commit/bb062c1588c324a1ce4cf01fd5e0780e83aaabe4

Check memory usage after  Neovim sourced the Vimscript files.
https://github.com/neovim/neovim/pull/12679

N/A patches for version.c:

vim-patch:8.2.0062: memory test is flaky on FreeBSD

Problem:    Memory test is flaky on FreeBSD.
Solution:   Add a short sleep before getting the first size.
https://github.com/vim/vim/commit/e7538ae997b3983d0c91a886a74ebacedd752164

vim-patch:8.2.0071: memory test often fails on Cirrus CI

Problem:    Memory test often fails on Cirrus CI.
Solution:   Allow for more tolerance in the upper limit.  Remove sleep.
https://github.com/vim/vim/commit/1832d12aea30f1533f3c461d9e1530d10f66b162

janlazo committed:
vim-patch:8.2.1254: MS-Windows: regexp test may fail if 'iskeyword' set wrongly

Problem:    MS-Windows: regexp test may fail if 'iskeyword' set wrongly.
Solution:   Override the 'iskeyword' value. (Taro Muraoka, closes vim/vim#6502)
https://github.com/vim/vim/commit/470adb827f3d9e6cf62f685738d2db216daf3738

teto commented:
I am very tempted to merge this. I wanted to mark the failing BSD test as pending before you tried to fix xD

janlazo committed:
vim-patch:8.2.1259: empty group in 'tabline' may cause using an invalid pointer

Problem:    Empty group in 'tabline' may cause using an invalid pointer.
Solution:   Set the group start position. (closes vim/vim#6505)
https://github.com/vim/vim/commit/f56c95fdad5af521887f8cd7bc15729b5355231d

janlazo committed:
vim-patch:8.2.1265: crash with EXITFREE when split() fails

Problem:    Crash with EXITFREE when split() fails.
Solution:   Restore 'cpoptions'.
https://github.com/vim/vim/commit/7d5e744162c1e971e5a863e89787cadc8e56051c

erw7 commented:
FYI https://github.com/neovim/neovim/pull/12665/commits/fb8d24cc12b3762ddca09c43fd2b2cb126c539ed fixes #12644.

janlazo committed:
vim-patch:8.2.1267: MS-Windows: tests may fail due to $PROMPT value

Problem:    MS-Windows: tests may fail due to $PROMPT value.
Solution:   Set $PROMPT for testing. (Taro Muraoka, closes vim/vim#6510)
https://github.com/vim/vim/commit/2690b5aed8b6b6070430b05dcae296cd9479c942

janlazo committed:
tests/terminal/tui: wait 1ms to avoid data race in FreeBSD

janlazo committed:
vim-patch:8.1.0862: no verbose version of character classes

Problem:    No verbose version of character classes.
Solution:   Add [:ident:], [:keyword:] and [:fname:]. (Ozaki Kiichi,
            closes vim/vim#1373)
https://github.com/vim/vim/commit/221cd9f4dd866503777b2fffa721c1403716ad63

janlazo committed:
vim-patch:8.1.0192: executing regexp recursively fails with a crash

Problem:    Executing regexp recursively fails with a crash.
Solution:   Move global variables into "rex".
https://github.com/vim/vim/commit/0270f38e1ae484c31a80c813a08691c47a207f1a

janlazo committed:
vim-patch:8.1.0194: possibly use of NULL pointer

Problem:    Possibly use of NULL pointer. (Coverity)
Solution:   Reset the re_in_use flag earlier.
https://github.com/vim/vim/commit/414998023fbff15cce20ef01a54d0366370ad8b6

janlazo committed:
vim-patch:8.1.1202: always get regexp debugging logs when building with -DDEBUG

Problem:    Always get regexp debugging logs when building with -DDEBUG.
Solution:   By default do not create regexp debugging logs. (Ken Takata)
https://github.com/vim/vim/commit/c2d09c9f2ce487a6971af9306e382d7b38805456

janlazo committed:
vim-patch:8.1.2340: quickfix test fails under valgrind and asan

Problem:    Quickfix test fails under valgrind and asan.
Solution:   Make sure long line does not overflow IObuff. (Dominique Pelle,
            closes vim/vim#5263)  Put back fix for large terminals. (Yegappan
            Lakshmanan, closes vim/vim#5264)
https://github.com/vim/vim/commit/a106e6cde682bda4ad10ed745acb51975fcb02e0

janlazo committed:
vim-patch:8.2.1292: AIDL filetype not recognized

Problem:    AIDL filetype not recognized.
Solution:   Add filetype detection. (Dominique Pellé, closes vim/vim#6533)
https://github.com/vim/vim/commit/2afc3b4f776a4fe2bb22d0a79e77012d79b4ec1e

janlazo committed:
vim-patch:8.2.1295: tests 44 and 99 are old style

Problem:    Tests 44 and 99 are old style.
Solution:   Convert to new style tests. (Yegappan Lakshmanan, closes vim/vim#6536)
https://github.com/vim/vim/commit/7d40b8a532fa2ce768f200ab118870d17875a7fe

janlazo committed:
options: fix 'isident' for Windows

Default value on Windows does not match the documented value.

janlazo committed:
vim-patch:8.1.0255: backup test fails when using shadow directory

Problem:    Backup test fails when using shadow directory.
Solution:   Remove check for "src".
https://github.com/vim/vim/commit/f8f88f89e12df516c1fac5851b504238ebc1d2d4

janlazo committed:
vim-patch:8.1.0264: backup tests fail when CWD is in /tmp

Problem:    Backup tests fail when CWD is in /tmp.
Solution:   Make 'backupskip' empty. (Christian Brabandt, closes vim/vim#3301)
https://github.com/vim/vim/commit/efe03738f69b1f63ea30226765db949539ee15f0

teto commented on test/functional/legacy/memory_usage_spec.lua:
```diff
@@ -143,16 +147,20 @@ describe('memory usage', function()
     local after = monitor_memory_usage(pid)
     for _ = 1, 3 do
       feed_command('so '..fname)
+      wait()
```
do we need all these `wait`s ?

janlazo replied:
`wait()` is needed after every `:source` so that the upper/lower tolerance values are the same as in Vim. I added `load_adjust` because the test can still fail on FreeBSD. Without the wait, the upper tolerance must be `time * 2` instead of `load_adjust(time * 1.5)`.

Question is why `:source` returns immediately.

teto commented on src/nvim/regexp_nfa.c:
```diff
@@ -6552,16 +6548,20 @@ static long nfa_regexec_both(char_u *line, colnr_T startcol,
 
   retval = nfa_regtry(prog, col, tm, timed_out);
 
+#ifdef REGEXP_DEBUG
```
why the additions of REGEXP_DEBUG ?

teto reviewed (approved):
if linting gets out of hand, I would be ok to merge with the issues. Now that it's done, great let's get it in.
Let me know if I can merge this.

janlazo commented:
It's ok to merge. I don't have anything to add at the moment.
